commands/topchannels.py

```diff
@@ -79,16 +83,22 @@ def get_recommended_channels(**kwargs):
                         channel_info["purpose"]["value"],
                         channel["num_members"],
                         float(channel_info["latest"]["ts"]),
+                        channel_info["latest"].get("subtype"),
                     )
                 )
 
     # now weight channels and return message
     potential_channels = sorted(
-        potential_channels, key=calc_channel_score, reverse=True
+        ((calc_channel_score(chan), chan) for chan in potential_channels), reverse=True
     )
 
-    for channel in potential_channels[:nr_channels]:
-        msg += f'{LIST_ICON} #{channel.name} ({channel.num_members} members, {seconds_since_last_post(channel) / 3600:.0f} hour(s) since last post): {channel.purpose if channel.purpose else "<Invest today and get an awesome description!>"}\n'
+    msg = MSG_BEGIN + "\n".join(
+        (
+            f'{LIST_ICON} #{channel.name} ({channel.num_members} members, {seconds_since_last_post(channel) / 3600:.0f} hour(s) since last post): {channel.purpose if channel.purpose else "<Invest today and get an awesome description!>"}'
```

Comment:
not very readable, I would make a template string constant at the top and `format` the variables in here

Reply:
Good call 👍